File: Data Structures & Algorithms/minimum-stack/submission-9.py
class MinStack:
    def __init__(self):
        self.l=[]
        self.minStack = []
        self.size=-1

    def push(self, val: int) -> None:

        self.l.append(val)
        if self.size < 0:
            self.minStack.append(val)
        else: 
            if val < self.minStack[self.size]:
                self.minStack.append(val)
            else:
                self.minStack.append(self.minStack[self.size])
        self.size+=1

    def pop(self) -> None:
        if self.size >= 0:
            self.l.pop()
            self.minStack.pop()
            self.size-=1

    def top(self) -> int:
        if self.size >=0: 
            return self.l[self.size] 

    def getMin(self) -> int:
        return self.minStack[self.size]

File: Data Structures & Algorithms/kth-largest-integer-in-a-stream/submission-9.cpp
class KthLargest {
public:

    std::priority_queue<int, std::vector<int>, std::greater<int>> minHeap;
    int kth;
    KthLargest(int k, vector<int>& nums) {
        kth = k;

        for (int n : nums) {
            minHeap.push(n);
        }


    }
    
    int add(int val) {
        minHeap.push(val);
        while (minHeap.size() > kth) {
            minHeap.pop();
        }

        return minHeap.top();
    }
};
